frontend/src/scenes/insights/insightLogic.ts

```diff
@@ -43,6 +46,66 @@ export const insightLogic = kea<insightLogicType>({
                 setAllFilters: (_, { filters }) => filters,
             },
         ],
+        /*
+        isFirstLoad determines if this is the first graph being shown after the component is mounted (used for analytics)
+        */
+        isFirstLoad: [
+            true,
+            {
+                setNotFirstLoad: () => false,
+            },
+        ],
+    }),
+    listeners: ({ values, actions }) => ({
+        setAllFilters: (filters) => {
+            actions.reportUsage(filters.filters)
+            actions.setNotFirstLoad()
+        },
+        reportUsage: async (filters: Record<string, any>, breakpoint) => {
+            // Reports `insight viewed` event
+            const { insight, display, interval, date_from, date_to } = filters
+            const properties: Record<string, any> = {
+                is_first_component_load: values.isFirstLoad,
+                insight,
+                display,
+                interval,
+                date_from,
+                date_to,
+                filters_count: filters.properties ? filters.properties.length : 0, // Only counts general filters (i.e. not per-event filters)
+                events_count: filters.events ? filters.events.length : undefined, // Number of event lines in insights graph; number of steps in funnel
+                actions_count: filters.actions ? filters.actions.length : undefined, // Number of action lines in insights graph; number of steps in funnel
+            }
+
+            properties.total_event_actions_count = (properties.events_count || 0) + (properties.actions_count || 0)
+
+            await breakpoint(5000)
+
+            // Custom properties for each insight
+            if (insight === 'TRENDS') {
+                properties.breakdown_type = filters.breakdown_type
+                properties.shown_as = filters.shown_as
+            } else if (insight === 'SESSIONS') {
+                properties.session_distribution = filters.session
+            } else if (insight === 'FUNNELS') {
+                properties.session_distribution = filters.session
+            } else if (insight === 'RETENTION') {
+                properties.period = filters.period
+                properties.date_to = filters.date_to
+                properties.retention_type = filters.retentionType
+                properties.same_retention_and_cohortizing_event =
+                    filters.returningEntity.events[0].id === filters.startEntity.events[0].id
+            } else if (insight === 'PATHS') {
+                properties.path_type = filters.path_type
+            }
+
+            const sanitizedProperties: Record<string, any> = {}
+            Object.entries(properties).map(([key, value]) => {
```

Comment:
Undefined properties already get dropped (this is default JSON.stringify behavior) - this is dead code.